src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -6097,21 +6097,29 @@ MethodTableBuilder::InitMethodDesc(
         break;
 
     case mcInstantiated:
-#ifdef EnC_SUPPORTED
-        if (fEnC)
-        {
-            // We reuse the instantiated methoddescs to get the slot
-            InstantiatedMethodDesc* pNewIMD = (InstantiatedMethodDesc*) pNewMD;
-            pNewIMD->SetupEnCAddedMethod();
-        }
-        else
-#endif // EnC_SUPPORTED
         {
             // Initialize the typical instantiation.
-            InstantiatedMethodDesc* pNewIMD = (InstantiatedMethodDesc*) pNewMD;
+            InstantiatedMethodDesc* pNewIMD = pNewMD->AsInstantiatedMethodDesc();
+
+            LoaderAllocator* pAllocator;
+            AllocMemTracker* pamTracker;
+            Module* pModule;
+            if (fEnC)
```

Comment:
What prevents the `!fEnC` path from working for EnC case?

If  MethodTableBuilder is not initialized enough, we should fix how it is initializated instead of this if-check.

Reply:
> If MethodTableBuilder is not initialized enough, we should fix how it is initializated instead of this if-check.

w00t! Thank you for considering that. Will do that.